gamemodes/terrortown/gamemode/server/sv_gamemsg.lua

```diff
@@ -233,48 +233,56 @@ end
 local cv_ttt_limit_spectator_chat = CreateConVar("ttt_limit_spectator_chat", "1", {FCVAR_ARCHIVE, FCVAR_NOTIFY})
 
 ---
--- Checks whether a @{Player} is able to listen to another @{Player}
+-- Returns whether or not the @{Player} can see the other @{Player}'s chat.
 -- @param string text
--- @param boolean team_only
--- @param Player listener
--- @param Player speaker
+-- @param boolean teamOnly
+-- @param Player reader
+-- @param Player sender
 -- @return boolean
 -- @hook
 -- @realm server
 -- @internal
-function GM:PlayerCanSeePlayersChat(text, team_only, listener, speaker)
-	if not IsValid(listener) then
+function GM:PlayerCanSeePlayersChat(text, teamOnly, reader, sender)
+	if not IsValid(reader) then
 		return false
 	end
 
-	if not IsValid(speaker) then
-		if IsEntity(speaker) then
-			return true
-		else
-			return false
-		end
+	if not IsValid(sender) and IsEntity(sender) then
```

Comment:
What if the sender is not "valid" and also is not an entity? Then the `sender:Team()` call would error, wouldn't it? 
Also where is the `return false`? Or rather does this break anything?

Reply:
I have no idea, this code was in there previously, I just removed the part Alf wanted to have removed

Reply:
But you removed the `IsEntity` part too. Maybe you should keep this as far as you don't know what it changes

Reply:
Yes, I had a discussion with saibotk and he wanted to have it removed. But maybe I misunderstood him there. Like I said to him, I don't know about the purpose of this codeelement.